day6/day6.rb

```diff
@@ -0,0 +1,71 @@
+require_relative './input'
+
+class Day6
+  def initialize(input)
+
+    @competition = Competition.new(input)
+  end
+
+  def part_one
+    @competition.number_of_wins_product
+  end
+
+  def part_two
+    @competition.number_of_wins_product
+  end
+end
+
+ class Competition
+  def initialize(input)
+
+    document = input.split("\n").reject{ |line| line == ""}
+    times = document[0].split(":")[1].split(" ")
+    distances = document[1].split(":")[1].split(" ")
+
+    @races = []
+    times.each_index do |i|
+      @races.push Race.new([times[i].to_i, distances[i].to_i])
+    end
```

Comment:
I'd also consider pulling this parsing logic out into a private helper method. You can call the helpers from the initializer, or lazy load them with a method called `races`. 

```ruby 
times = document[0].split(":")[1].split(" ").map(&:to_i) # this gets your values in the format you want right away. Easy to forget they're strings later. 
distances = document[1].split(":")[1].split(" ").map(&:to_i)

# try to avoid imperatively working with arrays. Its easier to map over things and process the elements directly rather than pushing to a placeholder array. 
@races = times.map.with_index |time, i| do
  Race.new(time, distance[i])
end
```